src/zenml/client.py

```diff
@@ -1040,7 +1034,7 @@ def update_role(
             name_id_or_prefix=name_id_or_prefix, allow_name_prefix_match=False
         )
 
-        role_update = RoleUpdateModel()
+        role_update = RoleUpdateModel(name=new_name or role.name)  # type: ignore
```

Comment:
Why do we need these type ignores for some of the update model creation now?

Reply:
There are some other non `optional` attributes, and because `mypy` seems unable to get the `@update_model`  decorator that makes every attribute actually optional

Reply:
But the `UserUpdateModel` should be the same right, it also uses the `@update_model` decorator and we seem to be able to create it without the ignore?

Btw not trying to say that you did anything wrong here, I'm just trying to figure out why mypy/the plugin is working so inconsistently for these update models (and also the `extra="allow"` thing).